File: pset1/cash/cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void)
{
    //variables//
    float quart = 25;
    int qcount = 0;

    float dime = 10;
    int dcount = 0;

    float nick = 5;
    int ncount = 0;

    float penn = 1;
    int pcount = 0;


    //negative value checker//
    float change;
    do
    {
        //user input//
        change = get_float("Change owed:\n");
    }
    while (change < 0);


    change = round(change * 100);


    //coin slotter//
    while (round(change) >= quart)
    {
        //Minus away quarters//
        change = change - quart;

        qcount++;
    }

    while (round(change) >= dime)
    {
        //Minus away dime//
        change = change - dime;

        dcount++;
    }

    while (round(change) >= nick)
    {
        //Minus away nickles//
        change = change - nick;

        ncount++;
    }

    while (round(change) >= penn && change != 0)
    {
        //Minus away pennies//
        change = change - penn;

        pcount++;
    }

    //coin counter//
    int coin = (qcount + dcount + ncount + round(pcount));
    printf("%i\n", coin);

}

File: pset6/cash/cash.py
from cs50 import get_float

#variables
quart = 25;
qcount = 0;

dime = 10;
dcount = 0;

nick = 5;
ncount = 0;

penn = 1;
pcount = 0;

#input
def input():
    while True:
        a = get_float("Change owed: ")
        if a > 0:
            break
    return a

a = input()
a = round(a * 100)

while a >= quart:
    a = a - quart
    
    qcount = qcount + 1
    
while a >= dime:
    a = a - dime
    
    dcount = dcount + 1
    
while a >= nick:
    a = a - nick
    
    ncount = ncount + 1
    
while a >= penn:
    a = a - penn
    
    pcount = pcount + 1
    
print(qcount + dcount + ncount + pcount)